### ISSUE TITLE
E2506 prevents using SourceSecurityGroupId for non-vpc security groups

### ISSUE THREAD

**Issue by andrew-nowak** [aws-cloudformation/cfn-lint#2310]
https://github.com/aws-cloudformation/cfn-lint/blob/4a7af2bd53a9ad1ccaba3a509437c53102ade522/src/cfnlint/rules/resources/ectwo/SecurityGroupIngress.py#L33-L40

I couldn't see any reason in the [cloudformation reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group-rule-1.html#cfn-ec2-security-group-rule-sourcesecuritygroupid) that this wouldn't be valid, and I was able successfully create the following stack, where SecurityGroupB seems to be correctly restricted to traffic from SecurityGroupA. I _think_ that this rule is incorrect, unless I'm missing something

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  SecurityGroupA:
    Type: AWS::EC2::SecurityGroup
    Properties:
      GroupDescription: Group A
      SecurityGroupIngress:
        - FromPort: 80
          ToPort: 80
          IpProtocol: tcp
          CidrIp: 0.0.0.0/0

  SecurityGroupB:
    Type: AWS::EC2::SecurityGroup
    Properties:
      GroupDescription: Group B
      SecurityGroupIngress:
        - FromPort: 80
          ToPort: 80
          IpProtocol: tcp
          SourceSecurityGroupId:
            Fn::GetAtt:
              - SecurityGroupA
              - GroupId
```

**Comment by sangchoigoat** [comment]
Yea it'd be nice to know what this rule is for. Also getting this error..

**Comment by kddejong** [comment]
This is an older rule and needs a refresh as there have been some changes from ec2-classic.  

Validated the following scenarios:
* `SourceSecurityGroupId` works for Default VPC (when `VpcId` is NOTspecified)
* `SourceSecurityGroupName` works  for Default VPC (when `VpcId` is NOTspecified)
* `SourceSecurityGroupName` doesn't work when VpcId is specified

Docs:

```
SourceSecurityGroupId
    The ID of the security group. You must specify either the security group ID or the security group name in the request. For security groups in a nondefault VPC, you must specify the security group ID.
    Required: No
    Type: String
    Update requires: [No interruption](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-no-interrupt)

SourceSecurityGroupName
    [Default VPC] The name of the source security group. You must specify either the security group ID or the security group name. You can't specify the group name in combination with an IP address range. Creates rules that grant full ICMP, UDP, and TCP access.

    For security groups in a nondefault VPC, you must specify the group ID.

    Required: No
    Type: String
    Update requires: [No interruption](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-no-interrupt)
```
